test/jasmine/tests/plot_api_test.js

```diff
@@ -84,6 +84,21 @@ describe('Test plot api', function() {
                 expect(gd._transitionData._frames[2].name).toEqual('frame3');
             }).catch(fail).then(done);
         });
+
+        it('should emit afterplot event after plotting is done', function(done) {
```

Comment:
This test here fails on `master`.

I'm not exactly sure why though which might me related to https://github.com/plotly/plotly.js/issues/76

Consider:

```js
var gd = document.getElementById('graph')

Plotly.plot(gd, [{}])

gd.on('plotly_afterplot', () => {})
```

I suspect that on `master` the `Plotly.plot` promise is resolved before `gd.on('plotly_relayout, /* */)` is set. By tucking `gd.emit('plotly_afterplot')` is the `Promise.all` callback the `Plotly.plot` promise is resolved after `gd.on`.